packages/flame_studio/lib/src/core/component_tree.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame_studio/src/core/game_controller.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final componentTreeProvider =
+    StateNotifierProvider<ComponentTreeObserver, ComponentTreeState>((ref) {
+  final gameState = ref.watch(gameControllerProvider);
+  return ComponentTreeObserver(gameState.game as FlameGame?);
+});
+
+@immutable
+class ComponentTreeNode {
+  const ComponentTreeNode._(this.component, this.children);
+
+  factory ComponentTreeNode.fromComponent(Component component) {
+    return ComponentTreeNode._(
+      component,
+      component.hasChildren
+          ? [
+              for (final child in component.children)
+                ComponentTreeNode.fromComponent(child)
+            ]
+          : null,
+    );
+  }
+
+  final Component component;
+  final List<ComponentTreeNode>? children;
+  String get name => component.runtimeType.toString();
```

Comment:
```suggestion
  late final String name = component.runtimeType.toString();
```

Reply:
`late final` fields don't work with const constructors. Of course, the same property can be initialized directly from the constructor, but it seems to be more complicated than a simple getter.